tests/test_gene_analysis.py

```diff
@@ -0,0 +1,261 @@
+        # Set up mock returns for empty gene set
+        empty_results = {}
+        for mock_func in [mock_go_ora, mock_wikipathways_ora, mock_reactome_ora, mock_phenotype_ora,
+                          mock_indra_upstream_ora, mock_indra_downstream_ora]:
+            mock_func.return_value = empty_results
+
+        result = discrete_analysis(
+            {},
+            client=self.mock_client,
+            method='bonferroni',
+            alpha=0.05,
+            keep_insignificant=True,
+            minimum_evidence_count=1,
+            minimum_belief=0.5
+        )
+
+        # All result sets should be empty
+        for analysis_type in result.keys():
+            self.assertEqual(len(result[analysis_type]), 0)
+
+    @patch('src.indra_cogex.analysis.gene_analysis.go_ora')
+    @patch('src.indra_cogex.analysis.gene_analysis.wikipathways_ora')
+    @patch('src.indra_cogex.analysis.gene_analysis.reactome_ora')
+    @patch('src.indra_cogex.analysis.gene_analysis.phenotype_ora')
+    @patch('src.indra_cogex.analysis.gene_analysis.indra_upstream_ora')
+    @patch('src.indra_cogex.analysis.gene_analysis.indra_downstream_ora')
+    @patch('src.indra_cogex.client.enrichment.discrete.count_human_genes', return_value=20000)
+    def test_significant_results_only(self, mock_count_human_genes, mock_indra_downstream_ora, mock_indra_upstream_ora,
+                                      mock_phenotype_ora, mock_reactome_ora, mock_wikipathways_ora, mock_go_ora):
+        # Set up mock returns with varying p-values
+        mock_go_ora.return_value = {
+            'CURIE:001': {'name': 'Term 1', 'p_value': 0.001, 'adjusted_p_value': 0.005},
+            'CURIE:002': {'name': 'Term 2', 'p_value': 0.01, 'adjusted_p_value': 0.05},
+            'CURIE:003': {'name': 'Term 3', 'p_value': 0.05, 'adjusted_p_value': 0.25},
+            'CURIE:004': {'name': 'Term 4', 'p_value': 0.1, 'adjusted_p_value': 0.5},
+            'CURIE:005': {'name': 'Term 5', 'p_value': 0.5, 'adjusted_p_value': 1.0}
+        }
+
+        result = discrete_analysis(
+            self.test_genes,
+            client=self.mock_client,
+            method='bonferroni',
+            alpha=0.05,
+            keep_insignificant=False,
+            minimum_evidence_count=1,
+            minimum_belief=0.5
+        )
+
+        # Check that only significant results (adjusted_p_value <= 0.05) are kept
+        self.assertIn('go_results', result)
+        significant_results = result['go_results']
+        self.assertEqual(len(significant_results), 2, "Test: Significant results only: Unexpected number of results")
+        self.assertIn('CURIE:001', significant_results)
+        self.assertIn('CURIE:002', significant_results)
+        self.assertNotIn('CURIE:003', significant_results)
+        self.assertNotIn('CURIE:004', significant_results)
+        self.assertNotIn('CURIE:005', significant_results)
+
+
+if __name__ == '__main__':
+    unittest.main()
```

Comment:
Is this a duplicate or does it need to appear both here and at the end of the file?

Reply:
I did resolve this use but these is no commit of this file. When I resolved this issue and commited on this file it stated  
```zsh: parse error near `()'
zsh: parse error in command substitution
``` 
When i again went to stage and commit the file it stated `nothing to commit, working tree clean`